test/test612.js

```diff
@@ -0,0 +1,144 @@
+if(typeof exports === 'object') {
+	var assert = require("assert");
+	var alasql = require('..');
+	var fs = require("fs");
+}
+
+var test = '612'; // insert test file number
+
+describe('Test '+test+' - INTO CSV', function() {
+	
+	before(function(){
+		alasql('CREATE DATABASE test' + test);
+		alasql('USE test' + test);
+		alasql('CREATE TABLE one (a INT, b VARCHAR)');
+		alasql('INSERT INTO one VALUES (10, \'swoll\'),' +
+                        '(11, \'muscles\')');
+
+	});
+
+	after(function(){
+		alasql('DROP DATABASE test' + test);
+		fs.unlink('test612-0.csv', function(err) {  });
+		fs.unlink('test612-1.csv', function(err) {  });
+		fs.unlink('test612-2.csv', function(err) {  });
+		fs.unlink('test612-3.csv', function(err) {  });
+        });
+
+	it('With quote = \'\', single string value', function(){
+		alasql("SELECT 'swing' AS `colname` INTO CSV('test612-0', {quote:''})");
+		fs.open('test612-0.csv', 'r', function(err, fd) {
+			if (err) {
+				console.error(err);
+			}
+
+			var buf = new Buffer(1024);
+			var csvstr = '';
+			fs.read(fd, buf, 0, buf.length, 0, function(err, bytes) {
+				if (err) {
+					console.error(err);
+				}
+
+				if (bytes > 0) {
+					csvstr = buf.slice(0, bytes).toString();
+				}
+
+				assert(csvstr === 'colname\n"swing"');
+		
+				fs.close(fd, function(err) {
+					if (err) {
+						console.error(err);
+					}
+				});
+			});
+        	});
+	});
+
+	it('With quote = \'\', single multiword string value', function(){
+		alasql("SELECT 'swing out' AS `colname` INTO CSV('test612-1', {quote:''})");
+		fs.open('test612-1.csv', 'r', function(err, fd) {

```

Comment:
Dont worry, we will get there. 

- As the alasql call is async we need to givethe testframework a way of knowing when the call is done. We do that by calling the fist parameter of the callback function to the `it(...` function. Its not named at the moment. Lets call it `done`

````
it('With quote = \'\', single string value', function(done){
```

and then call it after the assert with `done()`

- The `then(` function does not hvae a second parameter, but return a promise where we can `catch` errors.

- Also, reading a file is expensive. If you need the data twice 

- The assert function can take a true/false (you get that by the `===`) but it works better if you give each side as a parameter - because then the test will output what was expected and what we got instead (making it easyer to debug)

In all I suggest the code looks like 

```js
it('With quote = \'\', single string value', function(done ){
	alasql.promise("SELECT 'swing' AS `colname` INTO CSV('test612-0', {quote:''})")
		.then(function(state) {
			var filecontent = fs.readFileSync('test612-0.csv', 'utf8');
			assert(filecontent,'colname\nswing\n');
			done();
		}).catch(err) {
			//console.error(err)
		}
	;
});
``` 

I dont get the `\ufeff` part of the code you are trying to assert